core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosSchedulerBackend.scala

```diff
@@ -135,26 +136,26 @@ private[spark] class MesosSchedulerBackend(
       command.setValue(s"cd ${basename}*; $prefixEnv ./bin/spark-class $executorBackendName")
       command.addUris(CommandInfo.URI.newBuilder().setValue(uri))
     }
-    val cpus = Resource.newBuilder()
-      .setName("cpus")
-      .setType(Value.Type.SCALAR)
-      .setScalar(Value.Scalar.newBuilder()
-        .setValue(scheduler.CPUS_PER_TASK).build())
-      .build()
-    val memory = Resource.newBuilder()
-      .setName("mem")
-      .setType(Value.Type.SCALAR)
-      .setScalar(
-        Value.Scalar.newBuilder()
-          .setValue(MemoryUtils.calculateTotalMemory(sc)).build())
-      .build()
-    MesosExecutorInfo.newBuilder()
+
+    val builder = MesosExecutorInfo.newBuilder()
+
+    val (resourcesAfterCpu, usedCpuResources) =
+      partitionResources(resources, "cpus", scheduler.CPUS_PER_TASK)
+
+    builder.addAllResources(usedCpuResources)
+
+    val (resourcesAfterMem, usedMemResources) =
+      partitionResources(resourcesAfterCpu, "mem", MemoryUtils.calculateTotalMemory(sc))
+
+    builder.addAllResources(usedMemResources)
+
```

Comment:
can you remove these random new lines if they don't have special meanings? (see https://github.com/databricks/scala-style-guide for what "special meaning" here means)